### PR TITLE
Improve docstrings for manager, registry and auth server settings

### PR DESCRIPTION
**Pull request by maffoo**
@ejeffrey

### CODE REVIEW

File: src/main/scala/org/labrad/manager/auth/AuthServer.scala [hunk 1]
```diff
@@ -104,9 +113,13 @@ class AuthServer(
...
     }
 
-    @Setting(id=103,
-             name="authenticate",
-             doc="Check password for the given user")
+    @Setting(id = 103,
+             name = "authenticate",
+             doc = """Check credentials using the given authentication method.
+                 |
+                 |Called with a string idicating which authentication method to use, and data which
```
Comment (ejeffrey) [inline src/main/scala/org/labrad/manager/auth/AuthServer.scala]: idicating -> indicating

File: src/main/scala/org/labrad/registry/Registry.scala [hunk 2]
```diff
@@ -157,9 +157,9 @@ extends LocalServer with Logging {
       newDir
     }
 
-    @Setting(id=10,
-             name="cd",
-             doc="Change the current directory")
+    @Setting(id = 10,
+             name = "cd",
+             doc = """Change the current directory.""")
```
Comment (ejeffrey) [inline src/main/scala/org/labrad/registry/Registry.scala]: Optionally takes a second boolean argument to create the directory if it does not exist.

Also, you should document the integer argument, including stating that it is deprecated if you want to do that.

File: src/main/scala/org/labrad/registry/Registry.scala [hunk 3]
```diff
@@ -215,38 +228,57 @@ extends LocalServer with Logging {
...
       changeListener = if (enable) Some((r.source, id)) else None
     }
 
-    @Setting(id=55,
-             name="Stream Changes",
-             doc="Requests notifications of all changes made in any directory")
+    @Setting(id = 55,
+             name = "Stream Changes",
+             doc = """Requests notifications of all changes made in any directory.
+                 |
```
Comment (ejeffrey) [inline src/main/scala/org/labrad/registry/Registry.scala:276]: Is this all changes anywhere, or all changes in any subdirectory of the current context?

File: src/main/scala/org/labrad/manager/Handlers.scala [hunk 4]
```diff
@@ -398,68 +440,119 @@ class ManagerImpl(id: Long, name: String, hub: Hub, stub: ManagerSupport, tracke
...
+  @Setting(id = 200,
+           name = "Data To String",
+           doc = """Convert data into its human-readable string representation.""")
   def dataToString(data: Data): String = data.toString
 
-  @Setting(id=201, name="String To Data", doc="Parse a string into labrad data.")
+  @Setting(id = 201,
+           name = "String To Data",
+           doc = """Parse a string from human-readable representation into labrad data.""")
   def stringToData(str: String): Data = Data.parse(str)
```
Comment (ejeffrey) [inline src/main/scala/org/labrad/manager/Handlers.scala:540]: I think we should deprecate this -- or at least state what we expect about the stability of the string representation.

File: src/main/scala/org/labrad/manager/auth/AuthServer.scala [hunk 5]
```diff
@@ -158,13 +183,22 @@ class AuthServer(
       auth.changePassword(username, oldPassword, newPassword)
     }
```
Comment (ejeffrey) [inline src/main/scala/org/labrad/manager/auth/AuthServer.scala:184]: Not part of this PR, but this looks like a bug.  oldpassword should not be required for admin users.

File: src/main/scala/org/labrad/manager/auth/AuthServer.scala (hunk 1 above)
Comment (maffoo) [inline src/main/scala/org/labrad/manager/auth/AuthServer.scala]: fixed.

File: src/main/scala/org/labrad/registry/Registry.scala (hunk 2 above)
Comment (maffoo) [inline src/main/scala/org/labrad/registry/Registry.scala]: fixed.

File: src/main/scala/org/labrad/registry/Registry.scala (hunk 3 above)
Comment (maffoo) [inline src/main/scala/org/labrad/registry/Registry.scala:276]: Currently, this is all changes anywhere. This was originally done to make it possible to create a remote registry that proxies another registry, without having to keep open contexts in every directory to be notified of all changes. However, with the multi-headed registry this is not really needed anymore.

Comment (maffoo) [inline src/main/scala/org/labrad/registry/Registry.scala:276]: Might be more useful if this were all changes in a subdirectory of the current path, and that wouldn't be too hard to change.

File: src/main/scala/org/labrad/manager/auth/AuthServer.scala (hunk 5 above)
Comment (maffoo) [inline src/main/scala/org/labrad/manager/auth/AuthServer.scala:184]: Good point; I'll make a separate change.

File: src/main/scala/org/labrad/registry/Registry.scala (hunk 3 above)
Comment (maffoo) [inline src/main/scala/org/labrad/registry/Registry.scala:276]: Added a note to clarify the current behavior.

File: src/main/scala/org/labrad/manager/Handlers.scala (hunk 4 above)
Comment (maffoo) [inline src/main/scala/org/labrad/manager/Handlers.scala:540]: Added a note to indicate that the string representation is subject to change and shouldn't be used for long-term storage.

**Comment by ejeffrey** [comment]
LGTM

File: src/main/scala/org/labrad/registry/Registry.scala (hunk 3 above)
Comment (ejeffrey) [inline src/main/scala/org/labrad/registry/Registry.scala:276]: Yeah, all changes underneath the current path would be more useful, and could easily replicate the current behavior just by sitting in the root directory.
